test/e2e/storage/csi_volumes.go

```diff
@@ -343,27 +265,28 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
 		}
 		for _, t := range tests {
 			test := t
+			f := framework.NewDefaultFramework("csiworkload")
+
 			It(test.name, func() {
-				By("Deploying mock CSI driver")
-				config := testsuites.TestConfig{
-					Framework: f,
-					Prefix:    "csi-workload",
-				}
+				cs := f.ClientSet
+				csics := f.CSIClientSet
+				ns := f.Namespace
 
-				driver = drivers.InitMockCSIDriver(config, test.driverExists, true, test.podInfoOnMountVersion)
-				driver.CreateDriver()
+				By("Deploying mock CSI driver")
+				driver = drivers.InitMockCSIDriver(test.driverExists, true, test.podInfoOnMountVersion)
+				config, testCleanup := driver.PrepareTest(f)
 
 				if test.driverExists {
-					defer destroyCSIDriver(csics, driver)
-					defer driver.CleanupDriver()
+					defer destroyCSIDriver(csics, config.GetUniqueDriverName())
+					defer testCleanup()
```

Comment:
shouldn't we want to cleanup the test whether the driver exists or not

Reply:
Yes. This was introduced in https://github.com/kubernetes/kubernetes/pull/74079 and I noticed it too when rebasing on top of that, then notified @jarrpa. There's now a fix pending in https://github.com/kubernetes/kubernetes/pull/74079... which has been merged. I'll need to rebase once more.